# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/ipc/CompositorOptions.h
# Commit: f37913e7cb92
# Full Hash: f37913e7cb9295617b7e5a7ac5f462f29df583b4
# Author: Matt Woodrow <mwoodrow@mozilla.com>
# Date: 2021-08-18 15:53:06
# Regressor Bug: 1726063
# File Overlap Count: 1
# Description:
#   Bug 1726063 - Remove support for non-WR compositor initialization. r=jrmuizel,aosmond
#   
#   Depends on D122796
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D122798
# ==============================================================================

diff -r 7398bc391379 -r f37913e7cb92 gfx/ipc/CompositorOptions.h
--- a/gfx/ipc/CompositorOptions.h	Tue Aug 17 23:19:44 2021 +0000
+++ b/gfx/ipc/CompositorOptions.h	Tue Aug 17 23:19:44 2021 +0000
@@ -31,27 +31,21 @@
   // This constructor needed for IPDL purposes, don't use it anywhere else.
   CompositorOptions()
       : mUseAPZ(false),
-        mUseWebRender(false),
         mUseSoftwareWebRender(false),
         mAllowSoftwareWebRenderD3D11(false),
         mAllowSoftwareWebRenderOGL(false),
         mUseAdvancedLayers(false),
         mInitiallyPaused(false) {}
 
-  CompositorOptions(bool aUseAPZ, bool aUseWebRender,
-                    bool aUseSoftwareWebRender)
+  CompositorOptions(bool aUseAPZ, bool aUseSoftwareWebRender)
       : mUseAPZ(aUseAPZ),
-        mUseWebRender(aUseWebRender),
         mUseSoftwareWebRender(aUseSoftwareWebRender),
         mAllowSoftwareWebRenderD3D11(false),
         mAllowSoftwareWebRenderOGL(false),
         mUseAdvancedLayers(false),
-        mInitiallyPaused(false) {
-    MOZ_ASSERT_IF(aUseSoftwareWebRender, aUseWebRender);
-  }
+        mInitiallyPaused(false) {}
 
   bool UseAPZ() const { return mUseAPZ; }
-  bool UseWebRender() const { return mUseWebRender; }
   bool UseSoftwareWebRender() const { return mUseSoftwareWebRender; }
   bool AllowSoftwareWebRenderD3D11() const {
     return mAllowSoftwareWebRenderD3D11;
@@ -79,7 +73,7 @@
   }
 
   bool operator==(const CompositorOptions& aOther) const {
-    return mUseAPZ == aOther.mUseAPZ && mUseWebRender == aOther.mUseWebRender &&
+    return mUseAPZ == aOther.mUseAPZ &&
            mUseSoftwareWebRender == aOther.mUseSoftwareWebRender &&
            mAllowSoftwareWebRenderD3D11 ==
                aOther.mAllowSoftwareWebRenderD3D11 &&
@@ -91,7 +85,6 @@
 
  private:
   bool mUseAPZ;
-  bool mUseWebRender;
   bool mUseSoftwareWebRender;
   bool mAllowSoftwareWebRenderD3D11;
   bool mAllowSoftwareWebRenderOGL;